azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/CallingViewModel.kt

```diff
@@ -205,6 +217,27 @@ internal class CallingViewModel(
         updateLobbyOverlayDisplayedState(state.callState.callingStatus)
     }
 
+    private fun switchCameraOperation() {
+
+        val currentCameraID =
+            store.getCurrentState().localParticipantState.cameraState.deviceSelection.selectedCameraID
+        val listOfCameras =
+            store.getCurrentState().localParticipantState.cameraState.deviceSelection.cameras
+        val selectedCameraIndex = listOfCameras.keys.indexOf(currentCameraID)
+
+        if (selectedCameraIndex == listOfCameras.size - 1) {
```

Comment:
You can use a % operator here to only have one dispatch.

e.g. nextCameraIdx = (listOfCameras.keys.indexOf(currentCameraID) + 1) % listOfCameras.size